provider/kubernetes/kubernetes.go

```diff
@@ -515,8 +522,8 @@ func equalPorts(servicePort v1.ServicePort, ingressPort intstr.IntOrString) bool
 	return false
 }
 
-func shouldProcessIngress(ingressClass string) bool {
-	return ingressClass == "" || ingressClass == "traefik"
+func (p *Provider) shouldProcessIngress(ingressClass string) bool {
+	return ingressClass == "" || ingressClass == p.IngressClass
```

Comment:
I'm not sure if this expresses the desirable behavior. Here is what we should do from my understanding:

- If the user did **not** specify the `p.IngressClass` parameter, we need to process an Ingress object if the Ingress object annotation is either missing, has the empty value, or the `traefik` value. (This matches the current behavior where users cannot specify `p.IngressClass`.)
- If the user **did** specify the `p.IngressClass` parameter, we should only process an Ingress object if the Ingress object annotation value matches `p.IngressClass` exactly.

The current implementation seems to always process Ingress objects which are missing the annotation or come with an empty value.

See also how Nginx does it [here](https://github.com/nginxinc/kubernetes-ingress/tree/master/examples/multiple-ingress-controllers#using-multiple-ingress-controllers) and [here](https://github.com/kubernetes/ingress-nginx#running-multiple-ingress-controllers). I think we want to mimic that behavior.

Reply:
Also sorry -- I think I explained this part incorrectly on Slack. :( The logic is somewhat tricky here...